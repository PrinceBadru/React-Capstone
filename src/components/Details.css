.details-container {
  background-color: red;
}

.details-header {
  display: flex;
  padding: 11px;
  justify-content: center;
  background-color: red;
}

.img-cont {
  width: 10%;
  height: auto;
}

.logo-img {
  width: 100%;
}

.header-inner {
  color: #fff;
  display: flex;
  flex-direction: column;
  gap: 10px;
  align-items: stretch;
  align-content: flex-start;
  justify-content: center;
}

.divider-header {
  background-color: red;
  padding: 12px;
  align-items: start;
}

.details-list {
  display: flex;
  flex-direction: column;
  list-style: none;
  color: #fff;
  padding: 1.1px;
}

.details-item {
  border: 1px solid;
  padding: 20px;
}

@media screen and (max-width: 375px) {
  .logo-img {
    width: 70px;
  }
}
