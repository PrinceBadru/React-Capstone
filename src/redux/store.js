import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import cryptoReducer from './Crypto/CryptoSlice';

const middleware = [...getDefaultMiddleware(), thunk];

const store = configureStore({
  reducer: {
    crypto: cryptoReducer,
  },
  middleware,
});

export default store;
