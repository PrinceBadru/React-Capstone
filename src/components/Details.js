import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
import { fetchCrypto } from '../redux/Crypto/CryptoSlice';
import './Details.css';

const Details = () => {
  const dispatch = useDispatch();
  const { name } = useParams();
  const cryptoDetailsArray = useSelector((state) => state.crypto.data.coins);
  const cryptoDetails = cryptoDetailsArray
    ? cryptoDetailsArray.find((crypto) => crypto.name === name)
    : null;
  useEffect(() => {
    dispatch(fetchCrypto(name));
  }, [dispatch, name]);

  return (
    <div className="details-container">

      {cryptoDetails && (
        <>
          <div className="details-header">
            <div className="header-inner">
              <span className="detail">
                Rank:
                {' '}
                {cryptoDetails.rank}
              </span>
              <span className="detail">
                24Hour Volume:
                {' '}
                {cryptoDetails['24hVolume']}
              </span>
              <span className="detail">
                BTC Views:
                {' '}
                {cryptoDetails.btcPrice}
              </span>

            </div>
            <div className="img-cont">
              <img className="logo-img" src={cryptoDetails.iconUrl} alt="logo" />
            </div>
          </div>
          <div className="divider-header">
            <h3>
              Crypto Details for:
              {' '}
              {name}
            </h3>
          </div>
          <ul className="details-list">
            {Object.entries(cryptoDetails).map(([key, value], index) => (
              <li
                key={key}
                className="details-item"
                style={{
                  backgroundColor: index % 2 === 0 ? 'red' : 'red',
                }}
              >
                <strong>
                  {key}
                  :
                </strong>
                {' '}
                {value}
              </li>
            ))}
          </ul>
        </>
      )}
    </div>
  );
};

Details.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      name: PropTypes.string.isRequired,
    }),
  }).isRequired,
};

export default Details;
