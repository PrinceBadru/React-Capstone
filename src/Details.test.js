import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Details from './components/Details';
import store from './redux/store';

test('renders Details component correctly', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Router>
        <Details />
      </Router>
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
