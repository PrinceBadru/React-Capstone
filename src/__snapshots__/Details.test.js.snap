// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Details component correctly 1`] = `
<DocumentFragment>
  <div
    class="details-container"
  />
</DocumentFragment>
`;
